kernel/src/schema.rs

```diff
@@ -29,6 +29,17 @@ pub enum MetadataValue {
     Other(serde_json::Value),
 }
 
+impl MetadataValue {
+    pub(crate) fn as_string(&self) -> String {
+        match self {
+            MetadataValue::Number(n) => format!("{}", n),
+            MetadataValue::String(ref s) => s.clone(),
+            MetadataValue::Boolean(b) => format!("{}", b),
+            MetadataValue::Other(_) => panic!("Can't to_string other rn"),
```

Comment:
are we only using this in debug right now? might not want to panic?

Reply:
Yeah, should be fine. `serde_json::Value` implements `Display` so this is trivial. Done.